third_party/torch-mlir-dialects/test/tmtensor/invalid.mlir

```diff
@@ -0,0 +1,328 @@
+// RUN: iree-opt -split-input-file -verify-diagnostics %s
```

Comment:
It seems like these have been modified? We don't really want to be modifying the code here. Amongst other things it makes the update process much more difficult. I don't think it's really necessary to have the tests here at all, actually. They aren't going to be included in our own build (we don't do this for any other third party dependency). If we're concerned about making sure we've got a good version of these files, we can run the tests in that repository before importing. If you just want a smoke test to make sure the dialect is registered and such, you can throw one in the main IREE repo (we can discuss best location for that). But also the tests you have for the pass should be able to test the thing we really care about, I think.

Reply:
I asked for these tests... (FTR, I think having this in `third_party/...` is not a great idea, but lets leave that aside). These are copy of the ops in IREE. Without having these tests they will silently fail and will only be known when Torch-MLIR gets bumped. I dont think that is a good setup but we have gone back and forth so long on this, I am not going to litigate this further. I'll just register my dissent on this and go away.

Reply:
(I haven't looked at this specific example in detail but can take another more focused look after the conference this week)

Third party code must be included as pristine copies (or pristine copies + a clear list of patches) in third_party/.

We expect dependencies to be tested upstream. For the specific points where we use a dependency we test that behavior outside of third_party/. For example:
* https://github.com/google/iree/tree/main/iree/compiler/InputConversion/TOSA/test
* https://github.com/google/iree/tree/main/iree/test/e2e/tosa_ops

Reply:
Hey Scott, the tests I am trying to add are basic IR lit tests. Do you have any suggestions where to put them?